### ISSUE TITLE
CA1853 firing when another dictionary is used for ContainsKey

### ISSUE THREAD

**Issue by Octol1ttle** [dotnet/roslyn-analyzers#6377]
### Analyzer

**Diagnostic ID**: [CA1853](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1853): `Unnecessary call to 'Dictionary.ContainsKey(key)'`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 7 SDK](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.100](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

A false positive will occur if you use ContainsKey on a Dictionary that is other than the Dictionary the key is being removed from

### Steps To Reproduce

```csharp
var d1 = new Dictionary<string, string>();
var d2 = new Dictionary<string, string>();
foreach (var key in d1.Keys)
	if (!d2.ContainsKey(key)) d1.Remove(key); // False positive! Removing `d2.ContainsKey` will lead to every key being removed from `d1`!
```

**Comment by louis-z** [comment]
I could look into it.
